File: 91-aws/02-rest-api-mysql/src/main/java/com/in28minutes/rest/webservices/restfulwebservices/versioning/Name.java
package com.in28minutes.rest.webservices.restfulwebservices.versioning;

public class Name {
	private String firstName;
	private String lastName;

	public Name(String firstName, String lastName) {
		super();
		this.firstName = firstName;
		this.lastName = lastName;
	}

	public String getFirstName() {
		return firstName;
	}

	public String getLastName() {
		return lastName;
	}

	@Override
	public String toString() {
		return "Name [firstName=" + firstName + ", lastName=" + lastName + "]";
	}

}

File: 12-rest-api-kotlin/src/main/kotlin/com/in28minutes/rest/webservices/restfulwebservices/versioning/Name.kt
package com.in28minutes.rest.webservices.restfulwebservices.versioning
/**
 * A data class representing a person's name.
 *
 * @property firstName A read-only (`val`) property for the first name.
 * @property lastName A read-only (`val`) property for the last name.
 *
 * - This is a **data class**, which automatically provides:
 *   - `toString()`
 *   - `equals()`
 *   - `hashCode()`
 *   - `copy()`
 * - The constructor parameters (`val firstName`, `val lastName`) are automatically
 *   assigned as properties of the class.
 */
data class Name(
    val firstName: String,
    val lastName: String
)

 /**
  * Alternative approach using a regular class:
  * A standard class representing a person's name.
  *
  * @constructor Primary constructor with private properties (`firstName`, `lastName`).
  * These properties are not accessible outside the class directly.
  */
//class Name(
//    private val firstName: String,
//    private val lastName: String
//) {
//
//    // Explicit getters for serialization
//    fun getFirstName(): String = firstName
//    fun getLastName(): String = lastName
//
//    override fun toString(): String {
//        return "Name [firstName=$firstName, lastName=$lastName]"
//    }
//}
